src/containers/dynamicarray.d

```diff
@@ -158,6 +160,41 @@ struct DynamicArray(T, Allocator = Mallocator, bool supportGC = shouldAddGCRange
 	}
 
 	/**
+	 * Ensures sufficient capacity to accommodate `n` elements.
+	 */
+	void reserve(size_t n)
+	{
+		if (arr.ptr is null)
+		{
+			if (!n)
```

Comment:
Shouldn't this be the first thing that the function checks? (Also, please do not treat numbers as booleans)

Reply:
I expect that calling reserve with "n == 0" is uncommon. If the array is already initialized it's redundant with "n < array.length", but we could use the same check for both the initialized and uninitialized cases. I'll move the check up.